cou/steps/__init__.py

```diff
@@ -221,3 +230,61 @@ async def run(self) -> Any:
             return await self._task  # wait until task is completed
         except asyncio.CancelledError:  # ignoring asyncio.CancelledError
             logger.warning("Task %s was stopped unsafely.", repr(self))
+
+
+class ApplicationUpgradeStep(BaseStep):
```

Comment:
I think that `coro` should not be defined in BaseStep or we should by default set `None` for core in this class. So there will be no `ApplicationUpgradeStep` with `coro` defined. e,g. `ApplicationUpgradeStep(coro=my_coro())` will not be possible.